groovy update: regenerate

bilelmoussaoui opened the pull request:
Supersedes #13

bilelmoussaoui committed:
groovy update: regenerate

bilelmoussaoui committed:
gir: remove removed types

bilelmoussaoui committed:
generate missing events

bilelmoussaoui committed:
generate missing types

bilelmoussaoui committed:
generate missing builders

bilelmoussaoui committed:
gir: add missing manual types

bilelmoussaoui committed:
make it compile

this mostly removes the none existent any more apis
it also comments TimeCord as it's a record (couldn't manage to get it to work)

bilelmoussaoui commented:
This builds fine now, in the latest commit I have removed everything that called a function in sys that doesn't exists anymore. A more proper checking should happen later but it's beyond my objective of just updating gtk4-rs to the latest release

bilelmoussaoui committed:
regen with latest gir

sdroege commented on Gir.toml:
```diff
@@ -413,6 +417,15 @@ name = "Gdk.Monitor"
+manual_traits = ["PopupLayoutExtManual"]
+    [[object.function]]
+    name = "get_offset"
+    # invalid mutability, manual
```
`const = true` on the parameter in question doesn't help?

bilelmoussaoui replied:
It doesn't help here neither, though I forgot to add the manual get_offset. Should be done

sdroege replied:
Needs to get `(out)` annotations for these two parameters.

sdroege commented on Gir.toml:
```diff
@@ -432,3 +445,7 @@ manual_traits = ["SurfaceExtManual"]
         const = true
+    [[object.function]]
+    name = "translate_coordinates"
+    # mutability issue, manual
```
`const = true` on the parameter in question doesn't help?

bilelmoussaoui replied:
Nop it doesn't help :/  I keep getting
```
   --> src/auto/surface.rs:223:17
    |
223 |                 y,
    |                 ^ expected *-ptr, found `f64`
    |
    = note: expected raw pointer `*mut f64`
                      found type `f64`

 ```

sdroege replied:
Has wrong annotations in C: the parameters should be `(inout)`, then this should work here.

sdroege replied:
Please send MRs upstream for all these annotation problems and document that accordingly in the `Gir.toml` when implementing it manually :) Here the function should probably be `fn (self, x, y) -> (f64, f64)`

bilelmoussaoui replied:
I don't think it's needed here? as the function returns a boolean per the docs and it's just a mutability 
```
gboolean gdk_surface_translate_coordinates (GdkSurface *from,
                                            GdkSurface *to,
                                            double     *x,
                                            double     *y);
```

sdroege replied:
Why shouldn't it be needed? gobject-introspection can't know otherwise that `x` and `y` are `inout` parameters. You need to pass the coordinates in "from" space there, and get back the coordinates in "to" space

sdroege commented on Gir.toml:
```diff
@@ -88,7 +74,6 @@ manual = [
     "Gio.OutputStream",
     "GLib.Bytes",
     "GLib.Error",
-    "GLib.Priority",
```
Why?

bilelmoussaoui replied:
I don't exactly remember why, but reverted

sdroege reviewed (commented):
Looks good to me otherwise. Please rebase on the gitlab actions PR (after merging that one), and once the 3 comments are handled and the CI passes let's get this in.

bilelmoussaoui committed:
Gir: re-add Glib.Priority to manual

bilelmoussaoui committed:
popup layout: add get_offset

forgetten one and remove the manual set_offset
set_offset can be generated automatically

bilelmoussaoui commented:
So the only issue with the Gdk-4.gir file is the missing inout annotation for `gdk_popup_layout_get_offset` otherwise this merge request looks good to me.
I will open an issue in upstream gtk with the list from gdk/gsk/gtk 4

sdroege commented:
>  I will open an issue in upstream gtk with the list from gdk/gsk/gtk 4

Thanks

bilelmoussaoui commented:
I guess this one can go in now? :+1:

sdroege commented:
Looks good to me, yes

bilelmoussaoui commented:
Thanks for the review & help as always :) I will go ahead and update the gsk4 PR
